spec/lib/identity/parsers/gc_ids_spec.rb

```diff
@@ -106,6 +107,22 @@
         it 'returns a parsed sec id from the input xml object' do
           expect(subject[:sec_id]).to eq id
         end
+
+        context 'and the ids include one or multiple sec id regex matches' do
+          let(:id_object_one) { "#{id}^PN^200PROV^USDVA^A" }
+          let(:id_object_two) { '9988776655^PN^200PROV^USDVA^A' }
+          let(:ids) do
+            [
+              OpenStruct.new(attributes: { extension: id_object_one, root: root_oid }),
+              OpenStruct.new(attributes: { extension: id_object_two, root: root_oid })
+            ]
+          end
```

Comment:
I know we use `OpenStruct` in a lot of places but it's pretty non-performant, so much so that the [ruby docs](https://docs.ruby-lang.org/en/3.3/OpenStruct.html#class-OpenStruct-label-Caveats) say don't use it 🤣 so we should reconsider. In specs you can use `double` to accomplish the same thing
```ruby
let(:ids) do
  [
    double(attributes: { extension: id_object_one, root: root_oid }),
    double(attributes: { extension: id_object_two, root: root_oid })
  ]
end
```